### PR TITLE
timer-linux: prefer CLOCK_BOOTTIME if defined

### PR DESCRIPTION
**Pull request by N-R-K**
this is mainly just to unify the behavior between linux and BSDs where linux's `CLOCK_MONOTONIC{_RAW}` doesn't count time during system suspension. this likely doesn't matter much in practice given `CLOCK_MONOTONIC*` was used for a long time but it doesn't hurt unifying the behavior.

ref: https://github.com/mpv-player/mpv/pull/12764#discussion_r1375228918

### CODE REVIEW

**Comment by N-R-K** [comment]
On a side-note, it's pretty weird to call the file `timer-linux.c` when it's actually used in more than just linux platforms:

```py
if posix and not darwin
    timer_source = files('osdep/timer-linux.c')
endif
```

**Comment by orion1vi** [comment]
Why is incrementing time during suspension preferable? timer-darwin.c doesn't increment either.

**Comment by Dudemanguy** [comment]
I can't think of any possible case where this implementation would matter for us. In fact while playing, mpv actively disables suspension anyway. On the other hand, the semantical differences of the `_RAW` clock variant should in theory be better for mpv, so I see this change as a downgrade really.

**Comment by N-R-K** [comment]
> Why is incrementing time during suspension preferable?

Just to unify the behavior with BSDs, as said in the commit message.

> the semantical differences of the _RAW clock variant should in theory be better for mpv

I was under the impression that the BOOTTIME has similar guarantee as _RAW variant, but after reading the manpage more carefully that doesn't seem to be the case. So that's a good reason not to use it.

**Comment by kasper93** [comment]
Also I think all clock uses in mpv currently works fine with not incrementing clock during suspension. We don't have any absolute timing in fact and all of them are relative timeouts to current time point and those in fact would work less practicably if all timers would fire after resume instead just logically wait for the time (in mpv point of reference) elapses.

tl;dr we don't have any absolute time requirements.

That being said on Windows we actually include sleep time. On Darwin it does not include sleep and Linux also does not include sleep. But like I said, I don't think it matters to mpv in fact.

> [QueryPerformanceCounter](https://learn.microsoft.com/en-us/windows/win32/api/profileapi/nf-profileapi-queryperformancecounter) reads the performance counter and returns the total number of ticks that have occurred since the Windows operating system was started, including the time when the machine was in a sleep state such as standby, hibernate, or connected standby.
